### PR TITLE
Added the ability to list rows and delete rows by passing in row objects or row numbers.

### PR DESCRIPTION
**Pull request by NotLeet**
…Added the ability to list rows and delete rows by passing in row objects or row numbers. Wrote tests to confirm listing and deleting rows works as advertised.

### CODE REVIEW

**Comment by burnash** [comment]
Thanks for the PR. Please consider splitting the PR in two. There should be one PR per issue (even the small one).

File: gspread/client.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ class Client(object):
     :param http_session: (optional) A session object capable of making HTTP requests while persisting headers.
                                     Defaults to :class:`~gspread.httpsession.HTTPSession`.
 
-    >>> c = gspread.Client(auth=OAuthCredentialObject)
+    >>> c = gspread.Client(auth=OAuthCredentialObject)>
```
Comment (burnash) [inline gspread/client.py]: There's an extra '>' character at the end of the line.

**Comment by NotLeet** [comment]
I removed the fix for the attribute error so that can be addressed in a separate PR and got rid of the extra '>'

Is there anything else you'd like me to clean up before you merge it?

**Comment by synth3tk** [comment]
@burnash Is there anything else that this is waiting on?

**Comment by NotLeet** [comment]
I'd like to see this (or something like this) merged. This is a feature that a lot of people (including me) want and have wanted for quite some time.

You said that if I and others wanted this feature so badly, that I should write it and submit a PR. I did that and it doesn't seem to have done much good.

If there's a problem with the PR I can try and solve it, but if it sits here while you ignore it there's not a lot I can do.

**Comment by synth3tk** [comment]
@burnash Are there authorized contributors set up on this project? If not, it'd be nice if we could get that so the community could keep it going.

**Comment by pmochocki** [comment]
It would be really great if we could get this in.

**Comment by shopro** [comment]
Also very interested in seeing this feature. Any ETA?

**Comment by synth3tk** [comment]
Just checking in on this PR again. @burnash, if this were to get updated is it good to go?

**Comment by burnash** [comment]
@NotLeet Thank you again for the contribution. I'm really sorry for such a late reply. API v3 has been deprecated for quite a while. I have put all of the time I had for gspread into porting it to API v4.

I'm afraid the changes in this PR are not relevant anymore.

@synth3tk part of the functionality proposed in this PR (`Worksheet.delete_row()`) has been added in #337.

I'm closing this pull request.

**Pull request closed without merging**